src/QUICConnection.ts

```diff
@@ -410,73 +335,180 @@ class QUICConnection extends EventTarget {
+        this.secureEstablishedP, // This might reject with a relevant error
+        abortP, // This might abort for some other reason!
+      ]);
+    } catch (e) {
+      let e_ = e;
+      // This should only be true if we are infact aborted due to start timeout
+      if (ctx.signal.aborted) {
+        // No `QUICStream` objects could have been created, however quiche stream
+        // state should be cleaned up, and this can be done synchronously
+        for (const streamId of this.conn.readable() as Iterable<StreamId>) {
+          this.conn.streamShutdown(streamId, quiche.Shutdown.Read, 0);
         }
-        await this.send(mon);
-      },
+        for (const streamId of this.conn.writable() as Iterable<StreamId>) {
+          this.conn.streamShutdown(streamId, quiche.Shutdown.Write, 0);
+        }
+        // According to RFC9000, closing while in the middle of a handshake
+        // should use a transport error code `APPLICATION_ERROR`.
+        // For this library we extend this "handshake" phase to include the
+        // the TLS handshake too.
+        // This is also the behaviour of quiche when the connection is not
+        // in a "safe" state to send application errors (where `app` parameter is true).
+        // https://www.rfc-editor.org/rfc/rfc9000.html#section-10.2.3-3
+        this.conn.close(
+          false,
+          ConnectionErrorCode.ApplicationError,
+          Buffer.from('')
+        );
+        const localError = this.conn.localError()!;
+        e_ = new errors.ErrorQUICConnectionLocal(
+          'Failed to start QUIC connection due to start timeout',
+          {
+            data: localError,
+            cause: e
+          }
+        );
+        this.rejectSecureEstablishedP(e_);
+        this.dispatchEvent(
+          new events.EventQUICConnectionError({
+            detail: e_
+          })
+        );
+        this.dispatchEvent(
+          new events.EventQUICConnectionClose({
+            detail: {
+              type: 'local',
+              ...localError
+            }
+          })
+        );
+        // Send out the close!
+        await this.send();
+      }
+      // Wait for the connection to be fully closed
+      // It is expected that max idle timer will eventually resolve this
+      await this.closedP;
+      // Throw the augmented exception if it is augmented
+      // Otherwise throw the original
+      throw e_;
+    } finally {
+      ctx.signal.removeEventListener('abort', abortHandler);
+    }
+    if (this.config.keepAliveIntervalTime != null) {
+      this.startKeepAliveIntervalTimer(this.config.keepAliveIntervalTime);
+    }
+    this.addEventListener(
+      events.EventQUICConnectionError.name,
+      this.handleEventQUICConnectionError,
+    );
+    this.addEventListener(
+      events.EventQUICConnectionClose.name,
+      this.handleEventQUICConnectionClose,
+      { once: true }
     );
+    this.addEventListener(EventAll.name, this.handleEventQUICStream);
```

Comment:
We're registering `handleEventQuicStream` to all the `QUICConnection` events. This should only be added to the `QUICStream` events right?

Since the `handleEventQUICStream` is cloning all events, we're getting an infinite event loop and running into the max stack size.

I'm disabling for now, The handler should be applied to the `QUICStreams ` once they're created.

Reply:
Yes this should have been registered to the quic stream.

There are 2 places to do it, one on `newStream` and another on `processStreams`.

The removal of the handler will occur on the `EventQUICStreamDestroyed`.

Reply:
This is now fixed with b9d1212de609c88c0f2a9c595cb6037f178bae13.